### PR TITLE
#113 AwsDoc tests are missing

### PR DESCRIPTION
**Pull request by pinaf**
#113 AwsDoc tests are missing

### CODE REVIEW

**Comment by pinaf** [comment]
@karato please find a reviewer.

**Comment by karato** [comment]
Thanks, someone will review your pull request soon

**Comment by karato** [comment]
@alevohin do this review

File: src/test/java/com/nerodesk/om/aws/AwsDocTest.java [hunk 1]
```diff
@@ -49,4 +62,67 @@ public void conformsToEqualsHashCodeContract() {
...
+    @Test
+    public void reportsExistence() throws IOException {
+        final Bucket bucket = Mockito.mock(Bucket.class);
+        final Ocket ocket = Mockito.mock(Ocket.class);
+        Mockito.doReturn(true).when(ocket).exists();
+        final String label = "exists";
+        Mockito.doReturn(ocket).when(bucket).ocket(label);
+        MatcherAssert.assertThat(
+            new AwsDoc(bucket, label).exists(),
+            Matchers.equalTo(ocket.exists())
```
Comment (alevohin) [inline src/test/java/com/nerodesk/om/aws/AwsDocTest.java]: Could you explain why you don't use just `true` ?

**Comment by alevohin** [comment]
@pinaf great, thank you! Please see one small question.

File: src/test/java/com/nerodesk/om/aws/AwsDocTest.java (hunk 1 above)
Comment (pinaf) [inline src/test/java/com/nerodesk/om/aws/AwsDocTest.java]: @alevohin not that important. the value of true a few lines above is irrelevant if I compare the way I do now :)

**Comment by pinaf** [comment]
@alevohin answered

**Comment by alevohin** [comment]
@rultor merge @pinaf thank you!

**Comment by yegor256** [comment]
@pinaf I think that these tests would be much shorter and easier to understand if you would use built-in fake objects from `com.jcabi.s3.mock`: http://s3.jcabi.com/apidocs-0.9/com/jcabi/s3/mock/package-summary.html

**Comment by pinaf** [comment]
@yegor256 ok I'll take a look

**Comment by pinaf** [comment]
@yegor256 maybe for the first one (existence test), but the other use `Mockito.verify` which requires a Mockito mock.

**Comment by yegor256** [comment]
@pinaf but with fake classes you don't need `Mockito.verify()`. you just check what's in the bucket/ocket by reading from there. You don't need to know HOW the class communicated with S3 bucket/ocket - you want to know what is the result of the communication. It's a much healthier way to design tests, see http://www.yegor256.com/2014/09/23/built-in-fake-objects.html

**Comment by pinaf** [comment]
@yegor256 wow thanks for the advice! code readibility is night and day :) please take a look.

**Comment by yegor256** [comment]
@pinaf now it looks way better. these fake objects is the way to go. mocking sucks :)

**Comment by yegor256** [comment]
@rultor merge

**Comment by karato** [comment]
@alevohin Much obliged! I have added **17 mins** to your account in payment "54204771", 47 hours and 46 mins spent

review comments (c=2) added as a bonus

added +17 to your rating, now it is equal to [+1962](http://www.netbout.com/b/35105?open=rating)

**Comment by karato** [comment]
@rultor deploy

**Pull request merged**